theano/gpuarray/opt.py

```diff
@@ -389,17 +389,18 @@ def apply(self, fgraph):
             if new_o.type != o.type:
                 assert isinstance(o.type, tensor.TensorType)
                 assert isinstance(new_o.type, GpuArrayType)
-                assert new_o.owner
```

Comment:
great finding. If we add back the assert, is there a test failing? Just to be sure we don't break this later by adding it back.